cli/tests/e2e/test_baseline.py

```diff
@@ -621,6 +642,40 @@ def test_renamed_file(git_tmp_path, snapshot, new_name):
     }, "the old path should be gone now"
 
 
+def test_match_based_key(git_tmp_path, snapshot):
+    # Test that no findings in head reports no findings even if
+    # findings in baseline
+    foo = git_tmp_path / "foo.py"
+    foo.write_text(f"logger.log('foo')\n")
+
+    # Add baseline finding
+    subprocess.run(["git", "add", "."], check=True, capture_output=True)
+    _git_commit(1)
+    base_commit = subprocess.check_output(
+        ["git", "rev-parse", "HEAD"], encoding="utf-8"
+    ).strip()
+
+    # Add head finding
+    foo.write_text(f"logger.log('changed text')\n")  # Change just the content of foo
+    subprocess.run(["git", "add", "."], check=True, capture_output=True)
+    _git_commit(2)
+
+    # Non-baseline scan should report 1 finding
+    output = run_match_based_scan()
+    assert output.stdout != ""
+    snapshot.assert_match(
+        output.stderr.replace(base_commit, "baseline-commit"), "error.txt"
```

Comment:
Sort of a side comment, but why do we use stderr? The scan is succeeding, no?

Reply:
That's a good question. My understanding is we direct non-result output to stderr so that it makes it easier to filter. I'm not certain, though